language/diem-framework/modules/Diem.move

```diff
@@ -485,97 +551,337 @@ module Diem {
         include AbortsIfNoCurrency<CoinType>;
     }
 
-    /// Publishes a `Preburn` resource under `account`. This function is
+    /// Publish an empty `PreburnQueue` resource under the Designated Dealer
+    /// dealer account `account`.
+    fun publish_preburn_queue<CoinType>(
+        account: &signer
+    ) {
+        let account_addr = Signer::address_of(account);
+        Roles::assert_designated_dealer(account);
+        assert_is_currency<CoinType>();
+        assert(
+            !exists<Preburn<CoinType>>(account_addr),
+            Errors::invalid_state(EPREBURN)
+        );
```

Comment:
is this asserting the resource deletion occurrred beforehand?

Reply:
Yep! This is making sure that either it was either previously removed, or the account is fresh and doesn't have this yet.